README


This archive contains the distribution Dist-Zilla-Plugin-EnsurePrereqsInstalled,
version 0.007:

  Ensure at build time that all prereqs, including developer, are satisfied

This software is copyright (c) 2014 by Karen Etheridge.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.020.



Changes
Revision history for Dist-Zilla-Plugin-EnsurePrereqsInstalled

0.007     2014-09-27 03:41:56Z
          - make tests work in the presence of Test::More 1.301001_050

0.006     2014-08-09 03:27:46Z (TRIAL RELEASE)
          - dump a few more diagnostics on test failure, to try to identify
            the problem with dev releases of Test::More

0.005     2014-07-22 15:17:05Z
          - "types" option now also available as "prereq_types", for clarity
          - fix test regexps that fail when Test::More is a dev release

0.004     2014-06-22 00:04:06Z
          - fix tests that fail with CPAN::Meta::Check 0.009

0.003     2014-06-12 20:40:33Z
          - now warning the user at build time if dynamic_config is set, as
            the prereq specification may be incomplete
          - added new "build_phase" config, for performing the check at
            release time rather than build

0.002     2014-05-03 22:57:44Z
          - bump declared prereq on CPAN::Meta::Prereqs

0.001     2014-04-26 22:39:39Z
          - Initial release.


Build.PL
# This Build.PL for Dist-Zilla-Plugin-EnsurePrereqsInstalled was generated by
# Dist::Zilla::Plugin::ModuleBuildTiny::Fallback 0.007
use strict;
use warnings;

if (eval 'use Module::Build::Tiny 0.038; 1')
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";

    # This section for Dist-Zilla-Plugin-EnsurePrereqsInstalled was generated by Dist::Zilla::Plugin::ModuleBuildTiny 0.007.
    use strict;
    use warnings;

    use 5.006;
    # use Module::Build::Tiny 0.038;
    Build_PL();
}
else
{
    $ENV{PERL_MB_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are running this file manually, in which case you need
to learn to first fulfill all configure requires prerequisites listed in
META.yml or META.json -- or use a cpan client to install this distribution.

You can also silence this warning for future installations by setting the
PERL_MB_FALLBACK_SILENCE_WARNING environment variable, but please don't do
that until you fix your toolchain as described above.

EOW
    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));


    # This section was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.020.
    use strict;
    use warnings;

    require Module::Build; Module::Build->VERSION(0.28);


    my %module_build_args = (
      "build_requires" => {},
      "configure_requires" => {
        "ExtUtils::MakeMaker" => 0,
        "Module::Build::Tiny" => "0.038"
      },
      "dist_abstract" => "Ensure at build time that all prereqs, including developer, are satisfied",
      "dist_author" => [
        "Karen Etheridge <ether\@cpan.org>"
      ],
      "dist_name" => "Dist-Zilla-Plugin-EnsurePrereqsInstalled",
      "dist_version" => "0.007",
      "license" => "perl",
      "module_name" => "Dist::Zilla::Plugin::EnsurePrereqsInstalled",
      "recommends" => {},
      "recursive_test_files" => 1,
      "requires" => {
        "CPAN::Meta::Check" => "0.007",
        "CPAN::Meta::Prereqs" => "2.132830",
        "CPAN::Meta::Requirements" => 0,
        "Dist::Zilla::Plugin::MetaResources" => "4.300039",
        "Dist::Zilla::Role::AfterBuild" => 0,
        "Dist::Zilla::Role::BeforeBuild" => 0,
        "Dist::Zilla::Role::BeforeRelease" => 0,
        "Dist::Zilla::Util::AuthorDeps" => 0,
        "Moose" => 0,
        "Moose::Util::TypeConstraints" => 0,
        "Path::Class" => 0,
        "namespace::autoclean" => 0,
        "perl" => "5.006",
        "strict" => 0,
        "warnings" => 0
      },
      "script_files" => [],
      "test_requires" => {
        "Dist::Zilla::Role::MetaProvider" => 0,
        "ExtUtils::MakeMaker" => 0,
        "File::Spec" => 0,
        "Module::Metadata" => 0,
        "Path::Tiny" => 0,
        "Test::DZil" => 0,
        "Test::Deep" => 0,
        "Test::Fatal" => 0,
        "Test::More" => 0,
        "if" => 0,
        "lib" => 0
      }
    );


    my %fallback_build_requires = (
      "Dist::Zilla::Role::MetaProvider" => 0,
      "ExtUtils::MakeMaker" => 0,
      "File::Spec" => 0,
      "Module::Metadata" => 0,
      "Path::Tiny" => 0,
      "Test::DZil" => 0,
      "Test::Deep" => 0,
      "Test::Fatal" => 0,
      "Test::More" => 0,
      "if" => 0,
      "lib" => 0
    );


    unless ( eval { Module::Build->VERSION(0.4004) } ) {
      delete $module_build_args{test_requires};
      $module_build_args{build_requires} = \%fallback_build_requires;
    }

    my $build = Module::Build->new(%module_build_args);

    $build->create_build_script;
}


Makefile.PL
# This Makefile.PL for Dist-Zilla-Plugin-EnsurePrereqsInstalled was generated by
# Dist::Zilla::Plugin::MakeMaker::Fallback 0.013
# and Dist::Zilla::Plugin::MakeMaker::Awesome 0.27.
# Don't edit it but the dist.ini and plugins used to construct it.

use strict;
use warnings;

BEGIN {
my %configure_requires = (
    'ExtUtils::MakeMaker' => '0',
    'Module::Build::Tiny' => '0.038',
);

my @missing = grep {
    ! eval "require $_; $_->VERSION($configure_requires{$_}); 1"
} keys %configure_requires;

if (not @missing)
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";
}
else
{
    $ENV{PERL_MM_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

If you're installing manually, please retrain your fingers to run Build.PL
when present instead.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are doing something overly clever, in which case you
should consider setting the 'prefer_installer' config option in CPAN.pm, or
'prefer_makefile' in CPANPLUS, to 'mb" and '0' respectively.

You can also silence this warning for future installations by setting the
PERL_MM_FALLBACK_SILENCE_WARNING environment variable.

EOW

    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));
}
}

use 5.006;
use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Ensure at build time that all prereqs, including developer, are satisfied",
  "AUTHOR" => "Karen Etheridge <ether\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "Module::Build::Tiny" => "0.038"
  },
  "DISTNAME" => "Dist-Zilla-Plugin-EnsurePrereqsInstalled",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Dist::Zilla::Plugin::EnsurePrereqsInstalled",
  "PREREQ_PM" => {
    "CPAN::Meta::Check" => "0.007",
    "CPAN::Meta::Prereqs" => "2.132830",
    "CPAN::Meta::Requirements" => 0,
    "Dist::Zilla::Plugin::MetaResources" => "4.300039",
    "Dist::Zilla::Role::AfterBuild" => 0,
    "Dist::Zilla::Role::BeforeBuild" => 0,
    "Dist::Zilla::Role::BeforeRelease" => 0,
    "Dist::Zilla::Util::AuthorDeps" => 0,
    "Moose" => 0,
    "Moose::Util::TypeConstraints" => 0,
    "Path::Class" => 0,
    "namespace::autoclean" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Dist::Zilla::Role::MetaProvider" => 0,
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Module::Metadata" => 0,
    "Path::Tiny" => 0,
    "Test::DZil" => 0,
    "Test::Deep" => 0,
    "Test::Fatal" => 0,
    "Test::More" => 0,
    "if" => 0,
    "lib" => 0
  },
  "VERSION" => "0.007",
  "test" => {
    "TESTS" => "t/*.t"
  }
);

my %FallbackPrereqs = (
  "CPAN::Meta::Check" => "0.007",
  "CPAN::Meta::Prereqs" => "2.132830",
  "CPAN::Meta::Requirements" => 0,
  "Dist::Zilla::Plugin::MetaResources" => "4.300039",
  "Dist::Zilla::Role::AfterBuild" => 0,
  "Dist::Zilla::Role::BeforeBuild" => 0,
  "Dist::Zilla::Role::BeforeRelease" => 0,
  "Dist::Zilla::Role::MetaProvider" => 0,
  "Dist::Zilla::Util::AuthorDeps" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "Module::Metadata" => 0,
  "Moose" => 0,
  "Moose::Util::TypeConstraints" => 0,
  "Path::Class" => 0,
  "Path::Tiny" => 0,
  "Test::DZil" => 0,
  "Test::Deep" => 0,
  "Test::Fatal" => 0,
  "Test::More" => 0,
  "if" => 0,
  "lib" => 0,
  "namespace::autoclean" => 0,
  "strict" => 0,
  "warnings" => 0
);

unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
